### PR TITLE
perf(ivy): avoid repeat global state accesses in i18n instructions

### PR DESCRIPTION
**Pull request by crisbeto**
Removes repeat global state accesses from the i18n instructions in the cases where we have the information available already.

### CODE REVIEW

File: packages/core/src/render3/i18n.ts [hunk 1]
```diff
@@ -656,36 +656,36 @@ export function ɵɵi18nPostprocess(
...
 
 /**
  * See `i18nEnd` above.
  */
-function i18nEndFirstPass(tView: TView) {
-  const viewData = getLView();
+function i18nEndFirstPass(lView: LView, tView: TView) {
   ngDevMode && assertEqual(
-                   viewData[BINDING_INDEX], viewData[TVIEW].bindingStartIndex,
+                   lView[BINDING_INDEX], lView[TVIEW].bindingStartIndex,
```
Comment (petebacondarwin) [inline packages/core/src/render3/i18n.ts]: Can this be:

```suggestion
                   lView[BINDING_INDEX], tView.bindingStartIndex,
```

File: packages/core/src/render3/i18n.ts [hunk 2]
```diff
@@ -711,9 +711,8 @@ function createDynamicNodeAtIndex(
 }
 
 function readCreateOpCodes(
-    index: number, createOpCodes: I18nMutateOpCodes, icus: TIcu[] | null,
-    viewData: LView): number[] {
-  const renderer = getLView()[RENDERER];
+    index: number, createOpCodes: I18nMutateOpCodes, viewData: LView): number[] {
```
Comment (petebacondarwin) [inline packages/core/src/render3/i18n.ts]: Rename `viewData` to `lView` for consistency?

File: packages/core/src/render3/i18n.ts [hunk 3]
```diff
@@ -711,9 +711,8 @@ function createDynamicNodeAtIndex(
...
+  const renderer = viewData[RENDERER];
```
Comment (ocombe) [inline packages/core/src/render3/i18n.ts]: not sure if that's exactly the same, depending on the nesting?

Comment (crisbeto) [reply packages/core/src/render3/i18n.ts]: From what I could tell it's the same. I also tried adding a `console.log(viewData === getLView())` and it came back true for all of the i18n acceptance tests.

**Comment by crisbeto** [comment]
The feedback has been addressed @petebacondarwin.

**Review by petebacondarwin — APPROVED** [review]
I don't have enough understanding of the innards of ivy to know if caching the `LVIEW` rather than calling `getLView()` could break. But otherwise LGTM

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM

**Comment by AndrewKushnir** [comment]
- [VE Presubmit](https://test.corp.google.com/ui#id=OCL:272085278:BASE:272085297:1569884273674:ebb44ccd)
- [Ivy Presubmit](http://test/OCL:272085278:BASE:272085544:1569884350750:1bae4511)

**Review by ocombe — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
FYI, VE and Ivy presubmits are successful. Thank you.

**Pull request closed without merging**